libdlib/CMakeLists.txt

```diff
@@ -9,38 +9,38 @@ find_package(catkin REQUIRED)
 #test -e ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_SHARE_DESTINATION}/${PROJECT_NAME}/INSTALLED ||
 message(STATUS "Project source dir: ${PROJECT_SOURCE_DIR}")
 add_custom_target(
-    	build_dlib187 ALL
-    		COMMAND  cmake -E chdir ${PROJECT_SOURCE_DIR} make -f Makefile.tarball && touch ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_SHARE_DESTINATION}/${PROJECT_NAME}/INSTALLED 
+      build_dlib187 ALL
+        COMMAND  cmake -E chdir ${PROJECT_SOURCE_DIR} make -f Makefile.tarball && touch ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_SHARE_DESTINATION}/${PROJECT_NAME}/INSTALLED 
     )
 message(STATUS "done tarball")
 
 set(libraries "dlib")
 foreach(library ${libraries})
-	add_custom_target(
-		copy_${library} ALL
-		COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_SOURCE_DIR}/build/dlib-18.17/dlib/build/lib${library}.a ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_LIB_DESTINATION}
-    	COMMENT "Copying ${PROJECT_SOURCE_DIR}/common/build/lib${library}.a to ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_LIB_DESTINATION}"
-	)
+  add_custom_target(
+    copy_${library} ALL
+    COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_SOURCE_DIR}/build/dlib-18.17/dlib/build/lib${library}.a ${CATKIN_DEVEL_PREFIX}/${CATKIN_GLOBAL_LIB_DESTINATION}
```

Comment:
It depends on the package that is linking against this library. If the package (let's say people_detection) is trying to link statically we should copy `.a`. In order to support dynamic and static linking we should copy both libs (static and dynamic)